src/createReducedStream.ts

```diff
@@ -0,0 +1,54 @@
+import { BehaviorSubject, Observable, Subscription, Subject } from 'rxjs';
+import { ActionStream } from './types/helpers';
+import { RegisteredReducer, combineReducers } from './reducer';
+import { defaultErrorSubject } from './internal/defaultErrorSubject';
+import { markName } from './internal/markers';
+import { tag } from 'rxjs-spy/operators';
+
+export class ReducedStream<State> extends BehaviorSubject<State> {
+  subscription: Subscription;
+
+  constructor(initialState: State, reducedState$: Observable<State>) {
+    super(initialState);
+
+    this.subscription = reducedState$.subscribe(this);
+  }
+}
+
+/**
+ * Create a reduced state stream
+ *
+ * A reduced state stream is a stream that scans over an action stream and other
+ * stream to build up a state. It is eternally subscribed and always exposes
+ * it latest value like a Behaviour subject.
+ *
+ * ```
+ * const myState$ = createReducedStream(
+ *   'myState$',
+ *   initialState,
+ *   reducers,
+ *   action$
+ * );
+ *
+ * myState$.value === initialState // Will be true
+ * ```
+ *
+ * @param name The name of the stream, used for placing a marker and spy tag
+ * @param initialState The initial state of the stream
+ * @param reducers The reducers that build up the stream state
+ * @param action$ The action stream the action reducers should reduce over
+ */
+export const createReducedStream = <State>(
+  name: string,
+  initialState: State,
+  reducers: RegisteredReducer<State, any>[],
+  action$: ActionStream,
```

Comment:
**Is it intuitive to have `action$` as the last required argument?**

The argument list here is somewhat long, but I think it's better to have them as normal arguments than to put them in an object. Besides I believe that once we merge this, we should not attempt to change the signature without making a breaking change, so we can reconsider from scratch if we need to add more arguments later.

Reply:
But why not named arguments? The order doesn't really reveal what the arguments are, except perhaps the name?

Reply:
I guess this is just a matter of opinion or preference. I think with well named variables it is very easy to understand something like:
```ts
const myState$ = createReducedStream(
  'myState$',
  initialState,
  reducers,
  action$
);
```
And that an object / named arguments just adds more overhead. Especially considering that the ordering will not be consistent.
```ts
const myState$ = createReducedStream({
  name: 'myState$',
  reducers,
  action$,
  seed: initialState,
});
```

Reply:
Yes, with well named variables it works.